src/main/kotlin/net/atos/zac/app/audit/converter/RESTHistorieRegelConverter.kt

```diff
@@ -4,36 +4,46 @@
  */
 package net.atos.zac.app.audit.converter
 
-import jakarta.enterprise.inject.Any
-import jakarta.enterprise.inject.Instance
 import jakarta.inject.Inject
+import net.atos.client.zgw.brc.model.generated.Besluit
+import net.atos.client.zgw.drc.model.generated.EnkelvoudigInformatieObject
+import net.atos.client.zgw.drc.model.generated.Gebruiksrechten
+import net.atos.client.zgw.shared.model.ObjectType.BESLUIT
+import net.atos.client.zgw.shared.model.ObjectType.BESLUIT_INFORMATIEOBJECT
+import net.atos.client.zgw.shared.model.ObjectType.ENKELVOUDIG_INFORMATIEOBJECT
+import net.atos.client.zgw.shared.model.ObjectType.GEBRUIKSRECHTEN
+import net.atos.client.zgw.shared.model.ObjectType.OBJECT_INFORMATIEOBJECT
 import net.atos.client.zgw.shared.model.audit.AuditTrailRegel
 import net.atos.client.zgw.shared.model.audit.AuditWijziging
+import net.atos.zac.app.audit.converter.besluiten.AuditBesluitConverter
+import net.atos.zac.app.audit.converter.documenten.AuditEnkelvoudigInformatieobjectConverter
+import net.atos.zac.app.audit.converter.documenten.AuditGebruiksrechtenWijzigingConverter
 import net.atos.zac.app.audit.model.RESTHistorieRegel
 
-class RESTHistorieRegelConverter {
-
-    @Inject
-    @Any
-    lateinit var wijzigingConverterInstance: Instance<AbstractAuditWijzigingConverter<out AuditWijziging<*>>>
+class RESTHistorieRegelConverter @Inject constructor(
+    private val auditEnkelvoudigInformatieobjectConverter: AuditEnkelvoudigInformatieobjectConverter
+) {
 
     fun convert(auditTrail: List<AuditTrailRegel>): List<RESTHistorieRegel> =
         auditTrail.sortedByDescending { it.aanmaakdatum }
             .flatMap { convert(it) }
             .toList()
 
+    @Suppress("UNCHECKED_CAST")
```

Comment:
Previously in https://github.com/infonl/dimpact-zaakafhandelcomponent/pull/1455/files#diff-990d49a95cecbfe522f927302aef0960c9986fdfb2672c2d784b50953b7038f0L13